### PR TITLE
Check return value of pgp_sig_write.

### PR DESCRIPTION
**Pull request by dewyatt**
This should fail in CI as it depends on https://github.com/riboseinc/rnp/issues/364 being fixed first.

### CODE REVIEW

**Comment by flowher** [comment]
LGTM. Code is correct and I would propose comment failing test and merge this change.

File: src/lib/pgp-key.c [hunk 1]
```diff
@@ -604,10 +605,13 @@ pgp_add_selfsigned_userid(pgp_key_t *key, const uint8_t *userid)
     pgp_sig_end_hashed_subpkts(sig);
 
     if (!pgp_setup_memory_write(NULL, &sigoutput, &mem_sig, 128)) {
-        (void) fprintf(stderr, "can't setup memory write\n");
+        RNP_LOG("failed to set up memory write");
+        goto end;
+    }
+    if (!pgp_sig_write(sigoutput, sig, &key->key.seckey.pubkey, &key->key.seckey)) {
+        RNP_LOG("failed to write sig");
         return false;
```
Comment (flowher) [inline src/lib/pgp-key.c]: Please also goto end here

Comment (flowher) [inline src/lib/pgp-key.c]: Should go to end and free memory before leaving

Comment (dewyatt) [inline src/lib/pgp-key.c]: Good catch

**Comment by dewyatt** [comment]
> I would propose temporarily comment failing test and merge this change (after it is fixed). Please don't remove failing test as it also correct (and useful).

I was thinking we could just leave this open and you could push to this branch when you've had time to figure a fix, that way we would have CI tests passing before merging.

**Comment by flowher** [comment]
Ok, works for me. I'll be looking at it today, so should be done rather quickly

**Comment by flowher** [comment]
Code in ``rnp`` seems to be correct - all tests work correctly when botan is compiled with rfc6979 module disabled (``./configure --disable-mod=rfc6979``). Short term it could be a solution, nevertheless I would prefer we don't disable this module in long term.

I need to do quick review of RFC6979 before proposing patch to botan.

**Comment by ronaldtse** [comment]
Now that #364 is fixed should this be merged @dewyatt @flowher ?

**Comment by dewyatt** [comment]
Rebased on master and all tests are passing in CI, so IMO it's good to merge.

**Comment by flowher** [comment]
Seems this works after #370. I propose we merge it now

**Comment by ronaldtse** [comment]
Thank you guys!

**Pull request merged**